### ISSUE TITLE
No text-based suggesions?

### ISSUE THREAD

**Issue by bkhtrv** [vshaxe/vshaxe#638]
How do I enable suggestions for text with the Haxe extension enabled in VS Code?

![image](https://github.com/user-attachments/assets/db38df8a-0211-4dc9-8286-75058a964734)

**Comment by RblSb** [comment]
What do you mean?

**Comment by bkhtrv** [comment]
> What do you mean? Are you talking about completion or inlay hints? https://github.com/vshaxe/vshaxe/wiki/Completion https://github.com/vshaxe/vshaxe/wiki/Inlay-Hints

I’m not sure about the definitions, but this is what I mean:

![image](https://github.com/user-attachments/assets/0f0d9456-a620-47e1-a709-1dd4c7deb1c7)

I want the hint system to suggest the string “teststringlol” when I start typing “te”. It works when the Haxe plugin is disabled:

![image](https://github.com/user-attachments/assets/f37261c6-84b2-4554-b346-5a4849ed2ef0)

**Comment by bkhtrv** [comment]
So... you're telling me that vshaxe does not support text hints? :shocked:

**Comment by RblSb** [comment]
I don't see the point in having to have strings completion, rather you can accidentally trigger it and make a mistake instead of writing a new string. Is typescript or any other lang has this feature? I don't know how to enable it in other languages.

**Comment by bkhtrv** [comment]
> I don't see the point in having to have strings completion, rather you can accidentally trigger it and make a mistake instead of writing a new string. Is typescript or any other lang has this feature? I don't know how to enable it in other languages.

Well, when I open a JavaScript file in VS Code, string completion is enabled by default. This feature is also found in other major editors like Notepad++ and Sublime. For some reason, though, vshaxe messes with VS Code settings and doesn't include string completion, which is pretty useful for working with Maps, Reflect, and stuff like that.

**Comment by AlexHaxe** [comment]
I can see some usefulness in having some sort of suggestions when working with strings, although repetition of string literals usually is an indicator of something needing a refactoring, e.g. by using constants or enums.

I don't think Haxe supports completion requests on strings, so I guess that would have to be petitioned to the compiler people?!

**Comment by kLabz** [comment]
> I don't think Haxe supports completion requests on strings, so I guess that would have to be petitioned to the compiler people?!

I don't think so. It should be up to the editor (for example I do have completion in haxe strings for my other completion providers (file path, etc.)).

**Comment by AlexHaxe** [comment]
I guess if you only wanted string completion in the same file then that could be done by vshaxe.

I was thinking more about completing strings from all over your project which would require someone having access and knowing about all of your project's string literals, not only the ones currently opened in editor windows. though I don't know how useful that would be.
